### Periderm_ATACSeq

---------------

**Scripts for periderm_ATACSeq paper published in..**



####General workflow for ATAC-seq data process
- Environment setup (Ubuntu 18.0 or MacOS 10.12)
- QC using FastQC
- Trim the raw sequencing data using Trimmomatic (v0.36): (example paired-end raw read: C1_1.fq.gz, C1_2.fq.gz; dump unpaired output)
```
java -jar trimmomatic-0.38.jar PE -phred33 C1_1.fq.gz C1_2.fq.gz C1_1_paired.fq.gz C1_1_unpaired.fq.gz C1_2_paired.fq.gz C1_2_unpaired.fq.gz ILLUMINACLIP:NexteraPE-PE.fa:2:30:10:8:TRUE LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:5
```

- Mapping with Bowtie2 (example using trimmed paired reads, mapped to mm10 genome build);removal of PCR duplicates using picard tools; convert from sam output to bam output


- Size selection using custom scripts (located in /ATACSeq_Script/)

* * 1) size_selection.py (using deeptools as environment) to generate samfiles of NucleosomeFreeRegions and monoNucleosomeRegions

* * 2) SB_conversion.sh to generate bamfiles of nucleosomeFreeRegions


- bigWig outputs using deeptools: generate_bw_mm10.sh (for mouse data), generate_bw_hg19.sh (for human data) generate_bw_zv9.sh (for zebrafish data)
- peakCalling using MACS2: shift_and_peak_calling_zv9.sh (for zebrafish data); shift_and_peak_calling_mm10.sh (for mouse data); shift_and_peak_calling_hg19.sh (for human data)
#### Example script for plot

example scripts for plot

#### Scripts for Figure 1

**Name** ./Figure_1/

Figure 1B: matrix for heatmap was generated by seqMiner and plot using R (example script: heatmap.R)

Figure 1C and D: using matrix for Figure 1B and plot by R density plot (example script density_plot.R)

Figure 1H: scatter plot using R (example script scatter_plot.R)



**Comments** Used for generating heatmap for Figure 1B


